internal/store/image.go

```diff
@@ -111,78 +138,66 @@ func (b *imageStore) List() ([]reference.Reference, error) {
 	return references, nil
 }
 
-// Remove removes a bundle from the bundle store.
-func (b *imageStore) Remove(ref reference.Reference, force bool) error {
-	if id, ok := ref.(ID); ok {
-		refs := b.refsMap[id]
-		if len(refs) == 0 {
-			return fmt.Errorf("no such image %q", reference.FamiliarString(ref))
-		} else if len(refs) > 1 {
-			if force {
-				var failures *multierror.Error
-				toDelete := append([]reference.Reference{}, refs...)
-				for _, r := range toDelete {
-					if err := b.doRemove(r); err != nil {
-						failures = multierror.Append(failures, err)
-					}
-				}
-				return failures.ErrorOrNil()
-			}
-			return fmt.Errorf("unable to delete %q - App is referenced in multiple repositories", reference.FamiliarString(ref))
-		}
-		ref = refs[0]
-	}
-	return b.doRemove(ref)
-}
-
-func (b *imageStore) doRemove(ref reference.Reference) error {
-	path, err := b.storePath(ref)
+func (b *imageStore) listIDs() ([]string, error) {
+	f, err := os.Open(filepath.Join(b.path, "contents", "sha256"))
 	if err != nil {
-		return err
-	}
-	if _, err := os.Stat(path); os.IsNotExist(err) {
-		return errors.New("no such image " + reference.FamiliarString(ref))
+		return nil, err
 	}
-	b.refsMap.removeRef(ref)
-
-	if err := os.RemoveAll(path); err != nil {
-		return nil
+	defer f.Close()
+	ids, err := f.Readdirnames(-1)
+	if err != nil {
+		return nil, err
 	}
-	return cleanupParentTree(path)
+	return ids, nil
 }
 
-func cleanupParentTree(path string) error {
-	for {
-		path = filepath.Dir(path)
-		if empty, err := isEmpty(path); err != nil || !empty {
+// Remove removes a bundle from the bundle store.
+func (b *imageStore) Remove(ref reference.Reference, force bool) error {
+	if named, ok := ref.(reference.Named); ok {
+		named = reference.TagNameOnly(named)
+		id, err := b.store.Get(named)
+		if err != nil {
+			return err
+		}
+		_, err = b.store.Delete(named)
+		references := b.store.References(id)
+		if len(references) > 0 {
```

Comment:
What if `len(references) == 0`, we discard the error?

Reply:
this just mean we don't have tags for this, only a Digest